data_preprocessing.py

```diff
@@ -0,0 +1,122 @@
+import os
+import sys
+from segment import segment_audio, segment_video, preprocess_data
+from faster_whisper import WhisperModel
+import pandas as pd
+import csv
+from transformers import RobertaTokenizer
+import time
+import logging
+from logging.handlers import RotatingFileHandler
+
+#Logging 
+logging.basicConfig(
+    level=logging.INFO,  # Set the desired log level
+    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+    handlers=[
+        logging.StreamHandler(sys.stdout),  # log to console
+        RotatingFileHandler('output.log', maxBytes=1024*1024, backupCount=5)  # save to file
+    ]
+)
+logger = logging.getLogger(__name__)
+
+def Segment_audio_and_video(full_audio_file, full_video_file, output_dir):
+    '''
+  Given the full audio and video file, segment the audio and video and pass them to further preprocess functions.
+  Extract transcribtions from the audio. 
+  '''
+
+    model_size = "large-v2"
+
+    model = WhisperModel(model_size, device="cpu", compute_type="int8")
+
+    audio_in = full_audio_file #INPUT AUDIO
+    video_in = full_video_file #INPUT VIDEO
+
+    #TRANSCRIPTION
+    segments, info = model.transcribe(audio_in, beam_size=5, vad_filter=True)
+
+    print("Detected language '%s' with probability %f" % (info.language, info.language_probability))
+
+    #--- PARAMS ---#
+    #sampling rate
+    sr = 16000
```

Comment:
Parameters should in the end be configurable (typically). Although in this case, you might actually want to infer sampling rate from the audio file itself?